Add functionality to watch for transaction

tcharding opened the pull request:
First two commits are from: https://github.com/coblox/monero-harness-rs/pull/5 

Draft until that merges.

The wallet RPC call `check_tx_key` is used to get information about a transaction on the blockchain using the `tx_key`. The `tx_key` is the so called 'view key' i.e., a private key created for each transaction that allows one to view the transaction details.

Add requisite flags to `transfer` as required by `monero-wallec-rpc` so we get back the `tx_key` when doing a transfer. Implement `check_tx_key`.

Please see included test for example usage.

Commit:
Implement check_tx_key

The wallet RPC call `check_tx_key` is used to get information about a
transaction on the blockchain using the `tx_key`. The `tx_key` is the so called
'view key' i.e., a private key created for each transaction that allows one to
view the transaction details.

Add requisite flags to `transfer` as required by `monero-wallec-rpc` so we get
back the `tx_key` when doing a transfer. Implement `check_tx_key`.

bonomat commented on src/wallet.rs:
```diff
@@ -271,29 +297,73 @@ struct CreateWalletParams {
     account_index: u32,
+    // Destinations to receive XMR:
     destinations: Vec<Destination>,
+    // Set a priority for the transaction. Accepted Values are: 0-3 for: default, unimportant,
+    // normal, elevated, priority.
+    priority: u32,
+    // Number of outputs from the blockchain to mix with (0 means no mixing).
+    mixin: u32,
+    // Number of outputs to mix in the transaction (this output + N decoys from the blockchain).
+    ring_size: u32,
+    //  Number of blocks before the monero can be spent (0 to not add a lock).
+    unlock_time: u32,
+    // Return the transaction key after sending.
```
It looks like the default values in the `new` function have a specific meaning. Since those fields are optional, can we leave them out and let monero-wallet-rpc decide what's best?

tcharding replied:
These ones are not default according to the doc (assuming I'm reading it correctly). Priority could be interpreted as defaulting to 0 though.


https://web.getmonero.org/resources/developer-guides/wallet-rpc.html#transfer

bonomat replied:
I meant that the hardcoded values below look like they were chosen consciously. Imho they don't matter for our usecase and we can just let the monero-walelt chose for us.

tcharding replied:
Ok, tried and tested. You are right, even though the docs do not list these as optional the wallet does indeed pick sane values for us if they are left out. Have removed them in this PR now.

bonomat commented on tests/client.rs:
```diff
@@ -28,3 +28,43 @@ async fn init_accounts_for_alice_and_bob() {
+    let container = Container::new(&docker);
+    let cli = Client::new(container.monerod_rpc_port, container.wallet_rpc_port);
+
+    let balance_alice = 1000;
```
The problem is here: Alice simply does not have enough money: 
You've added a field to `TransferParams` called `priority` and default to `0`. I'm not sure how monero-wallet-rpc comes up with a fee based on this, but sending a transaction resulted in a fee of `336_780_000`. Obviously, she can't pay for this. Hence increase the value to 1 monero and it will work: 

```suggestion
    let balance_alice = 1_000_000_000_000;
```

I had a hunch that this was the error and simply increased the value. It worked, so I reduced it again until I saw this error on the logs: 

```
2020-09-23 22:39:03.019	E needed_money + min_fee > balance_subtotal. THROW EXCEPTION: error::not_enough_money
2020-09-23 22:39:03.020	W /data/monero.git/src/wallet/wallet2.cpp:9838:N5tools5error16not_enough_moneyE: not enough money, available = 0.000100000000, tx_amount = 0.000010000000
```

tcharding replied:
Nice work, I only increased it to 1_000_000 and never got this, well done. How did you get the fee amount (its not in the posted output above)?

bonomat replied:
I saw it in the logs of the test:  

```
transfer RPC response: {
  "id": "1",
  "jsonrpc": "2.0",
  "result": {
    "amount": 100,
    "fee": 336780000,
    "multisig_txset": "",
    "tx_blob": "",
    "tx_hash": "dd6a8c43e0edf8655cacb0e871bf2072112dd88391c4301ad582af8e8f53dcb6",
    "tx_key": "a9fb89f5191f9a5c59f025c9d19142fcabc1513bea48eaebf412557ce3649a05",
    "tx_metadata": "",
    "unsigned_txset": "",
    "weight": 1436
  }
}
```

Commit:
Remove TransferParams constants

We do not need to pass these parameters to the wallet, if left out the wallet
will pick sane values for us.
